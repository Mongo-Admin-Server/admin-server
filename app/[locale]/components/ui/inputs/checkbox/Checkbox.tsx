import styles from './checkbox.module.scss';

interface CheckboxProps {
  id?: string;
  label: string;
  value: boolean;
  disabled?: boolean;
  // eslint-disable-next-line no-unused-vars
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  error?: string;
  color?: string;
}

const Checkbox = ({
  id,
  label,
  value,
  disabled,
  onChange,
  error,
  color,
}: CheckboxProps) => {
  
  return (
    <div className={styles.container}>
      <label htmlFor={id} className={`${styles.label} ${color && styles[color]} ${value && styles.checked}`}>
        <input
          id={id}
          className={`${styles.input} ${error ? styles.inputError : ''} ${color && styles[color]}`}
          type="checkbox"
          disabled={disabled}
          checked={value}
          onChange={onChange}
        />
        {label}
      </label>
      {error && <p className={styles.error}>{error}</p>}
    </div>
  );
};

export default Checkbox;
