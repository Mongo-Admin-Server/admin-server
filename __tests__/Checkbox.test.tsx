/* eslint-disable no-undef */
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Checkbox from '@/app/[locale]/components/ui/inputs/checkbox/Checkbox';

const mock = {
  label: 'form checkbox',
  value: true
};

describe('Checkbox', () => {
  it('renders', () => {
    const handleChange = jest.fn();
    const { container } = render(
      <Checkbox 
        label={mock.label} 
        value={mock.value} 
        onChange={handleChange}
        disabled={false}
      />
    );

    expect(container).toMatchSnapshot();
  })
})