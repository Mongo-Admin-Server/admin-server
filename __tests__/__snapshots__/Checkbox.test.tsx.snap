// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders 1`] = `
<div>
  <div
    class="container"
  >
    <label
      class="label"
    >
      <input
        checked=""
        class="input "
        type="checkbox"
      />
      form checkbox
    </label>
  </div>
</div>
`;
